monoio-http-client/src/client/pool.rs

```diff
@@ -171,20 +218,28 @@ where
 
 impl<K, IO> ConnectionPool<K, IO>
 where
-    K: Hash + Eq + ToOwned<Owned = K>,
+    K: Hash + Eq + ToOwned<Owned = K> + Display,
 {
     pub fn get(&self, key: &K) -> Option<PooledConnection<K, IO>> {
-        if let Some(v) = unsafe { &mut *self.conns.get() }.mapping.get_mut(key) {
-            #[cfg(feature = "logging")]
-            tracing::debug!("connection got from pool");
-            return v.pop_front().map(|idle| PooledConnection {
-                key: Some(key.to_owned()),
-                codec: Some(idle.codec),
-                pool: Rc::downgrade(&self.conns),
-                reuseable: true,
-            });
+        let conns = unsafe { &mut *self.conns.get() };
+
+        match conns.mapping.get_mut(key) {
+            Some(v) => {
+                #[cfg(feature = "logging")]
+                tracing::debug!("connection got from pool for key: {:?} ", key.to_string());
+                v.pop_front().map(|idle| PooledConnection {
+                    key: Some(key.to_owned()),
+                    codec: Some(idle.codec),
+                    pool: Rc::downgrade(&self.conns),
+                    reuseable: true,
+                })
+            }
+            None => {
+                #[cfg(feature = "logging")]
+                tracing::debug!("no connection in pool for key: {:?} ", key.to_string());
```

Comment:
1. For debug usage, you should use trait Debug instead of Display.
2. Let's say you use Display, and here can be written as `tracing::debug!("no connection in pool for key: {key}"` instead.
3. `to_string` and format will bring an additional useless alloc here.